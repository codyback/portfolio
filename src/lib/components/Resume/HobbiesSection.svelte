<script lang="ts">
	import type { Hobby } from "@prisma/client";
  import Saos from "saos";
  
  export let hobbies: Hobby[];
</script>

<Saos once animation="{"from-right 0.5s cubic-bezier(0.230, 1.000, 0.320, 1.000) both"}">
  <section id="cvHobbies">
    <div class="mt-10 border-b-2 border-black pb-3">
      <h3 class="uppercase text-2xl font-bold text-right">Hobbies</h3>
    </div>

    <div class="border-t-4 border-black mt-0.5 pt-8">
      <p class="text-justify text-neutral-500">
        I enjoy taking on challenging projects as an opportunity to learn and grow, and am dedicated to delivering high-quality work.
      </p>

      <div class="mt-6 font-bold flex-initial flex-wrap flex-row">
        {#each hobbies as hobby (hobby.id)}
          <span class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase">
            {hobby.name}
          </span> 
        {/each}
      </div>
    </div>
  </section>
</Saos>
