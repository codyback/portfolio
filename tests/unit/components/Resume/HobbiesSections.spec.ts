import HobbiesSection from '@/lib/components/Resume/HobbiesSection.svelte';
import type { Hobby } from '@prisma/client';
import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';

const hobbies: Hobby[] = [
  {
    id: 'hobby-1',
    name: 'Hobby 1',
    url: 'hobby-1-url',
    homepageId: 'test',
  },
  {
    id: 'hobby-2',
    name: 'Hobby 2',
    url: 'hobby-2-url',
    homepageId: 'test',
  },
  {
    id: 'hobby-3',
    name: 'Hobby 3',
    url: 'hobby-3-url',
    homepageId: 'test',
  },
];

describe('CertificatesSection', () => {
  it('renders correctly with open false', () => {
    const { container } = render(HobbiesSection, { hobbies });

    expect(container).toMatchSnapshot();
  });
});
