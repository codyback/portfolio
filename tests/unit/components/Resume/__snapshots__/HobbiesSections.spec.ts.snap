// Vitest Snapshot v1

exports[`CertificatesSection > renders correctly with open false 1`] = `
<body>
  <div>
    <section
      class="mt-10 sm:mt-20"
      id="cvHobbies"
    >
      <div
        class="border-b-2 border-black dark:border-neutral-500 pb-3"
      >
        <h3
          class="uppercase text-2xl font-bold text-right"
        >
          Hobbies
        </h3>
      </div>
       
      <div
        class="border-t-4 border-black dark:border-neutral-500 mt-0.5"
      />
      <!--&lt;SectionHeading&gt;-->
       
      <div
        class="pt-8"
      >
        <p
          class="text-justify dark:text-neutral-300 text-neutral-500"
        >
          I enjoy taking on challenging projects as an opportunity to learn and
			grow, and am dedicated to delivering high-quality work.
        </p>
         
        <div
          class="mt-6 font-bold flex-initial flex-wrap flex-row"
        >
          <span
            class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase"
          >
            Hobby 1
             
          </span>
          <span
            class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase"
          >
            Hobby 2
             
          </span>
          <span
            class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase"
          >
            Hobby 3
             
          </span>
        </div>
      </div>
    </section>
    <!--&lt;HobbiesSection&gt;-->
  </div>
</body>
`;
